tests/xharness/Hardware/Device.cs

```diff
@@ -0,0 +1,144 @@
+﻿using System;
+using Xharness;
+
+namespace xharness.Hardware {
+	public class Device : IHardwareDevice {
+		public string DeviceIdentifier { get; set; }
+		public string DeviceClass { get; set; }
+		public string CompanionIdentifier { get; set; }
+		public string Name { get; set; }
+		public string BuildVersion { get; set; }
+		public string ProductVersion { get; set; }
+		public string ProductType { get; set; }
+		public string InterfaceType { get; set; }
+		public bool? IsUsableForDebugging { get; set; }
+		public bool IsLocked { get; set; }
+
+		public string UDID { get { return DeviceIdentifier; } set { DeviceIdentifier = value; } }
+
+		public string OSVersion => ProductVersion;
+
+		// Add a speed property that can be used to sort a list of devices according to speed.
+		public int DebugSpeed => InterfaceType?.ToLowerInvariant () switch
+		{
+			"usb" => 0,
```

Comment:
Please don't refactor code when moving it around, it makes everything much harder to review. For instance: the comments that explained why this order in particular disappeared.

Reply:
My bad, will re-add comment.